crates/matrix-sdk-ui/tests/integration/timeline/decryption.rs

```diff
@@ -0,0 +1,345 @@
+
+    let room_id = room_id!("!DovneieKSTkdHKpIXy:morpheus.localhost");
+    let event_factory = EventFactory::new().room(room_id).sender(&BOB);
+
+    let mock_server = MatrixMockServer::new().await;
+    let client = mock_server.client_builder().build().await;
+
+    // Set up the event cache store.
+    {
+        let event_cache_store = client.event_cache_store().lock().await.unwrap();
+
+        // The event cache contains one chunk as such:
+        //
+        // 1. a chunk of 1 item
+        //
+        // The item is an encrypted event! It has been stored before having a chance to
+        // be decrypted. Damn. We want to see if decryption will trigger automatically.
+        event_cache_store
+            .handle_linked_chunk_updates(
+                room_id,
+                vec![
+                    // chunk #1
+                    Update::NewItemsChunk {
+                        previous: None,
+                        new: ChunkIdentifier::new(0),
+                        next: None,
+                    },
+                    // … and its item
+                    Update::PushItems {
+                        at: Position::new(ChunkIdentifier::new(0), 0),
+                        items: vec![event_factory
+                            .event(RoomEncryptedEventContent::new(
+                                EncryptedEventScheme::MegolmV1AesSha2(
+                                    MegolmV1AesSha2ContentInit {
+                                        ciphertext: "\
+                                            AwgAEtABPRMavuZMDJrPo6pGQP4qVmpcuapuXtzKXJyi3YpEsjSWdzuRKIgJzD4P\
+                                            cSqJM1A8kzxecTQNJsC5q22+KSFEPxPnI4ltpm7GFowSoPSW9+bFdnlfUzEP1jPq\
+                                            YevHAsMJp2fRKkzQQbPordrUk1gNqEpGl4BYFeRqKl9GPdKFwy45huvQCLNNueql\
+                                            CFZVoYMuhxrfyMiJJAVNTofkr2um2mKjDTlajHtr39pTG8k0eOjSXkLOSdZvNOMz\
+                                            hGhSaFNeERSA2G2YbeknOvU7MvjiO0AKuxaAe1CaVhAI14FCgzrJ8g0y5nly+n7x\
+                                            QzL2G2Dn8EoXM5Iqj8W99iokQoVsSrUEnaQ1WnSIfewvDDt4LCaD/w7PGETMCQ"
+                                            .to_owned(),
+                                        sender_key: "DeHIg4gwhClxzFYcmNntPNF9YtsdZbmMy8+3kzCMXHA"
+                                            .to_owned(),
+                                        device_id: "NLAZCWIOCO".into(),
+                                        session_id: SESSION_ID.into(),
+                                    }
+                                    .into(),
+                                ),
+                                None,
+                            ))
+                            .event_id(event_id!("$ev0"))
+                            .into_utd_sync_timeline_event(),
+                          ],
+                    },
+                ],
+            )
+            .await
+            .unwrap();
```

Comment:
Instead of unwrapping everywhere, it would be really nice to move some of those comments into `expect()` calls. That way if the test fails, we have the description right in the shell that run the test, instead of searching for the source file.

Reply:
Yup, done!